src/sctools/test/test_bam.py

```diff
@@ -157,3 +194,71 @@ def test_split_succeeds_with_raise_missing_false_and_no_cr_barcode_passed(tagged
     os.remove(tagged_bam)  # clean up
     for f in glob.glob('test_output_*'):
         os.remove(f)
+
+
+def test_tag_sortable_records_sort_correctly():
```

Comment:
I find this test name slightly confusing since it's not sorting the records. Maybe rename this to `test_tag_sortable_record_comparison`?

Reply:
I think having more specific unit tests can help with debugging if there are breaking changes, e.g. a test that just checks that two records with different query names are sorted correctly.